dll/win32/imm32/precomp.h

```diff
@@ -93,7 +93,11 @@ static inline BOOL Imm32IsCrossProcessAccess(HWND hWnd)
             (DWORD_PTR)NtCurrentTeb()->ClientId.UniqueProcess);
 }
 
-#define ImeDpi_IsUnicode(pImeDpi) ((pImeDpi)->ImeInfo.fdwProperty & IME_PROP_UNICODE)
+#define ImeDpi_IsUnicode(pImeDpi)   ((pImeDpi)->ImeInfo.fdwProperty & IME_PROP_UNICODE)
+#define Imm32IsImmMode()            (g_psi && (g_psi->dwSRVIFlags & SRVINFO_IMM32))
+#define Imm32IsCiceroMode()         (g_psi && (g_psi->dwSRVIFlags & SRVINFO_CICERO_ENABLED))
+#define Imm32Is16BitMode() \
+    (((PTHREADINFO)NtCurrentTeb()->Win32ThreadInfo)->TIF_flags & TIF_16BIT)
```

Comment:
```suggestion
    (GetWin32ClientInfo()->dwTIFlags & TIF_16BIT)
```
The `THREADINFO` structure contents are inaccessible from usermode.

Reply:
I see.